crates/egui/src/widgets/plot/axis.rs

```diff
@@ -0,0 +1,297 @@
+    /// The second paramter is the maximum number of characters that fit into y-labels.
+    /// The second paramter of `formatter` is the currently shown range on this axis.
+    pub fn formatter(mut self, fmt: fn(f64, usize, &RangeInclusive<f64>) -> String) -> Self {
+        self.formatter = fmt;
+        self
+    }
+
+    fn default_formatter(tick: f64, max_digits: usize, _range: &RangeInclusive<f64>) -> String {
+        if tick.abs() > 10.0_f64.powf(max_digits as f64) {
+            let tick_rounded = tick as isize;
+            return format!("{:+e}", tick_rounded);
+        }
+        let tick_rounded = round_to_decimals(tick, max_digits);
+        if tick.abs() < 10.0_f64.powf(-(max_digits as f64)) && tick != 0.0 {
+            return format!("{:+e}", tick_rounded);
+        }
+        format!("{}", tick_rounded)
+    }
+
+    /// Specify axis label.
+    ///
+    /// The default is 'x' for x-axes and 'y' for y-axes.
+    pub fn label(mut self, label: impl Into<WidgetText>) -> Self {
+        self.label = label.into();
+        self
+    }
+
+    /// Specify maximum number of digits for ticks.
+    ///
+    /// This is considered by the default tick formatter and affects the width of the y-axis
+    pub fn max_digits(mut self, digits: usize) -> Self {
+        self.digits = digits;
+        self
+    }
+
+    /// Specify the placement of the axis.
+    pub fn placement(mut self, placement: Placement) -> Self {
+        self.placement = placement;
+        self
+    }
+
+    pub(super) fn thickness(&self) -> f32 {
+        match AXIS {
+            X_AXIS => {
+                if self.label.is_empty() {
+                    1.0 * LINE_HEIGHT
+                } else {
+                    3.0 * LINE_HEIGHT
+                }
+            }
+            Y_AXIS => {
+                if self.label.is_empty() {
+                    (self.digits as f32) * LINE_HEIGHT
+                } else {
+                    (self.digits as f32 + 1.0) * LINE_HEIGHT
+                }
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub(super) struct AxisWidget<const AXIS: usize> {
+    pub(super) range: RangeInclusive<f64>,
+    pub(super) hints: AxisHints<AXIS>,
+    pub(super) rect: Rect,
+    pub(super) transform: Option<PlotTransform>,
+    pub(super) steps: Vec<GridMark>,
+}
+
+impl<const AXIS: usize> AxisWidget<AXIS> {
+    /// if `rect` as width or height == 0, is will be automatically calculated from ticks and text.
+    pub(super) fn new(hints: AxisHints<AXIS>, rect: Rect) -> Self {
+        Self {
+            range: (0.0..=0.0),
+            hints,
+            rect,
+            transform: None,
+            steps: Vec::new(),
+        }
+    }
+}
+
+impl<const AXIS: usize> Widget for AxisWidget<AXIS> {
+    fn ui(self, ui: &mut Ui) -> Response {
+        // --- add label ---
+        let response = ui.allocate_rect(self.rect, Sense::click_and_drag());
```

Comment:
does the axis ever response to clicking and dragging, or should this be changed to `Sense::hover()` instead?